Source/KSCrash/Source/KSCrash/Recording/BSG_KSCrashReport.c

```diff
@@ -1543,77 +1549,99 @@ void bsg_kscrashreport_writeStandardReport(
-        bsg_kscrw_i_writeReportInfo(
+    {
+        bsg_kscrashreport_writeKSCrashFields(crashContext, writer, recordKSCrashFields);
+
+        if (crashContext->config.onCrashNotify != NULL) {
+            // NOTE: The blacklist for BSG_KSCrashField_UserAtCrash children in BugsnagEvent.m
+            // should be updated when adding new fields here
+
+            // Write handled exception report info
+            writer->beginObject(writer, BSG_KSCrashField_UserAtCrash);
+            if (crashContext->crash.crashType == BSG_KSCrashTypeUserReported) {
+                bsg_kscrashreport_writeBugsnagPayload(crashContext, writer);
+
+                if (recordKSCrashFields) {
+                    bsg_kscrashreport_writeOverrides(crashContext, writer);
+                }
+            }
+            { bsg_kscrw_i_callUserCrashHandler(crashContext, writer); }
+            writer->endContainer(writer);
+        }
+    }
+    writer->endContainer(writer);
+
+    bsg_ksjsonendEncode(bsg_getJsonContext(writer));
+
+    close(fd);
+}
+
+void bsg_kscrashreport_writeBugsnagPayload(const BSG_KSCrash_Context *crashContext,
+                                           const BSG_KSCrashReportWriter *writer) {
+    if (crashContext->crash.userException.eventOverrides != NULL) {
+        writer->addJSONElement(writer, BSG_KSCrashField_EventJson,
+                crashContext->crash.userException.eventOverrides);
+    }
+}
+
+void bsg_kscrashreport_writeOverrides(const BSG_KSCrash_Context *crashContext,
+                                      const BSG_KSCrashReportWriter *writer) {
+    if (crashContext->crash.userException.overrides != NULL) {
+        writer->addJSONElement(writer, BSG_KSCrashField_Overrides,
+                crashContext->crash.userException.overrides);
+    }
+    if (crashContext->crash.userException.handledState != NULL) {
+        writer->addJSONElement(writer, BSG_KSCrashField_HandledState,
+                crashContext->crash.userException.handledState);
+    }
+    if (crashContext->crash.userException.metadata != NULL) {
+        writer->addJSONElement(writer, BSG_KSCrashField_Metadata,
+                crashContext->crash.userException.metadata);
+    }
+    if (crashContext->crash.userException.state != NULL) {
+        writer->addJSONElement(writer, BSG_KSCrashField_State,
+                crashContext->crash.userException.state);
+    }
+    if (crashContext->crash.userException.config != NULL) {
+        writer->addJSONElement(writer, BSG_KSCrashField_Config,
+                crashContext->crash.userException.config);
+    }
+    writer->addIntegerElement(writer, BSG_KSCrashField_DiscardDepth,
+            crashContext->crash.userException.discardDepth);
+}
+
+void bsg_kscrashreport_writeKSCrashFields(BSG_KSCrash_Context *crashContext,
+                                          BSG_KSCrashReportWriter *writer, 
+                                          bool recordKSCrashFields) {
```

Comment:
If we pull the `writeReportInfo` up a level, there's no need to pass the param down and all the logic around which fields to write is nicely contained in one function.

Reply:
Amended as suggested.